endorsement/storeapi.proto

```diff
@@ -43,6 +63,62 @@ message OpenArgs {
 
 message CloseArgs {
 }
+message SwComponentID {
+    AttestationFormat type = 1;
+    google.protobuf.Struct id = 2;
+}
+// SwComponentInfo defines the structure of a single Software Component 
+// identified by a SoftwareComponent Identifer.
+message SwComponentInfo {
+    SwComponentID comp_id = 1;
+    google.protobuf.Struct comp_meta = 2; 
+}
+
+message AddSwComponentArgs {
+    repeated SwComponentInfo info = 1;
+}
+
+message GetSwComponentArgs {
+    SwComponentID comp_id = 1;
+}
+
+message GetSwComponentRsp {
+    google.protobuf.Struct comp_meta = 1;
+    int32 error_value = 2;
+    string error_detail = 3;
+}
+
+message TrustAnchorID {
+    AttestationFormat type = 1;
+
+    // Trust Anchor ID assumes a form based on the type of AttestationFormat
+    google.protobuf.Struct id = 2;
+}
+
+message TrustAnchorValue {
+    TaType type = 1;
+
+    google.protobuf.Struct value = 2;
+}
+
+message TrustAnchorInfo {
+    TrustAnchorID id = 1;
+    TrustAnchorValue value = 2;
+}
+
+message AddTrustAnchorArgs {
+    repeated TrustAnchorInfo trust_anchors = 1;
```

Comment:
What does it mean to Add multiple TAs in one go?  I am not sure I understand the use case for this.

Reply:
Agree, but what are the chances of failure use case you depicted and losing the entire payload versus the benefit of adding in single message.
Protobuf is precisely very good at such serialization! Hence I see a value in adding the interface like this!

Reply:
> Processing cost: Its fairly obvious that if the information is present to the sender, it is much more efficient to send it in one bundle against repeated messages of the same nature over the wire!

of course, at a micro-optimisation level there is no question.  But how that impacts on the overall processing is impossible to estimate at this point in time.  IOW, how do we know this is going to be a bottleneck?  And the answer is until we measure, we can't know.

> Interface correctness and ergonomics are not compromised laying out interface in this manner!

They are, as I have explained above.

> I am opposed to adding an Identical message : _AddManyTrustAnchors_ interface, as it duplicates the existing interface with no added value. if one trust anchor is needed, the count can be one, if more than one then the count can be much more!

Ditto.

> The same principle has been followed in Software Components also!

I don't think the analogy is not correct, because SwComponents (AFAIU) are bundled around the same "update" block, so it makes sense to cluster them and make them share fate within the same transaction.

> The interface is adding flexibility to set multiple items of identical type in the interface!

See above.